client/register.go

```diff
@@ -43,11 +43,28 @@ var registerTimeout = cmdRegister.Flag.Int("t", 5, "")
 const registerRecheckTime = 10 * time.Millisecond
 
 func runRegister(cmd *Command, args []string) {
-	if *registerKey == "" && *registerKeyFile == "" {
+	k := NewKeysFile(daemonFolder + daemonToRegister)
+	if *registerRemove && *registerKey == "" && *registerKeyFile == "" {
+		// Short circuit & handle `knox register -r`, which is expected to remove all keys
+		err := k.Lock()
+		if err != nil {
+			fatalf("There was an error obtaining file lock: %s", err.Error())
+		}
+		err = k.Overwrite([]string{})
+		if err != nil {
+			k.Unlock()
```

Comment:
log err here if it occurs?

Reply:
seems convention is to use fatalf / stdout for these errors.